### ISSUE TITLE
Adjust section level in included markdown files

### ISSUE THREAD

**Issue by olowo726** [DCsunset/pandoc-include#33]
Consider the following markdown

```
# Software components
## Platform

!include J1939Tp/J1939Tp.md
```

J1939Tp.md:
```
# J1939Tp

Implements the Transport Protocol (TP) 
.
.
.
```

Then I would like the heading J193Tp to be on level 3 in the merge output. "Software components" is level 1, "Platform" is level 2. The filter should know the heading level at the point of include and adjust heading levels in included file accordingly. This makes it possible to reuse included .md files in several documents.

**Comment by DCsunset** [comment]
I believe you could use the `incrementSection` option to specify how many levels you want to increment. The reason for not detecting it automatically is that it will make it impossible to add some level 1 sections at the end (like the appendix)

**Comment by olowo726** [comment]
incrementSection helps some but doesn't solve all cases. Consider for example nested includes. When writing a document which might be included the author doesn't know the current level and hence doesn't know desired increment.

How about incrementSection="auto"?

**Comment by olowo726** [comment]
Consider the case of document1.md which includes document2.md which
includes document3.md. The author of document2.md cannot know the
appropriate absolute level of document3.md since he/she doesn't know in
which context document1 will include document2.

Den fre 21 apr. 2023 kl 18:15 skrev DCsunset ***@***.***>:

**Comment by DCsunset** [comment]
Is it necessary for doc2 to know the the absolute level in doc1? `incrementSection` works by incrementing all the section levels. So as long as doc2 knows the how to set doc3's level relatively, it should work because doc1 can increment the levels of both doc2 and doc3 altogether

**Comment by olowo726** [comment]
You mean that the total increment sections in  doc3 will be the sum of
incrementSection from doc1 and doc2?

**Comment by DCsunset** [comment]
Yes I think so. It's based on the nature of recursive include.

**Comment by olowo726** [comment]
I've tested now and it works as you say. So current functionality works but I still think incrementSection="auto" would be a nice feature (saving me some thinking and make it easier to rearrange).

**Comment by DCsunset** [comment]
Yes I think that may help. However, it's hard to define the auto behaviour as there's no end boundary for sections.

For example, if we have the following markdown:

```md
# Sec 1

Section 1

# Sec 2

Section 2

!include file.md
```

What would you expect the incrementSection to be? It's reasonable to use either 0 or 1 here.

**Comment by olowo726** [comment]
I would use 1 since "Section 2" would be a normal text under hedline Sec 2

**Comment by DCsunset** [comment]
I see. It makes sense now. `auto` is a kinda confusing name as I originally thought you were suggesting a more intelligent approach. I would vote for `current` as it always needs to match with the current level.

**Comment by olowo726** [comment]
Ok
